integration/data_flow.md

```diff
@@ -0,0 +1,109 @@
+EXTENSION_TO_FILE_CLASS = {
+    '.fits.fz': 'ocs_archive.input.lcofitsfile.LcoFitsFile',
+    '.fits': 'ocs_archive.input.fitsfile.FitsFile',
+    '.tar.gz': 'ocs_archive.input.tarwithfitsfile.TarWithFitsFile',
+    '.pdf': 'ocs_archive.input.file.DataFile'
+}
+```
+
+This mapping can be overridden using the `FILETYPE_MAPPING_OVERRIDES` environment variable if the default is not sufficient or needs to be changed.
+
+
+#### Filestore
+
+A set of classes which represent storage backends exists within the `storage` module of the `ocs_archive` library. 
+
+The base [`FileStore` class](https://github.com/observatorycontrolsystem/ocs_archive/blob/main/ocs_archive/storage/filestore.py#L20) is intended to represent a data product storage backend, and can be sub-classed and extended to work with many different storage solutions.
+
+For example, the [`S3Store`](https://github.com/observatorycontrolsystem/ocs_archive/blob/main/ocs_archive/storage/s3store.py#L23) contains customizations to the base `FileStore` class to allow observatory data products to be stored in [Amazon S3](https://aws.amazon.com/s3/). If an observatory needs to use a different cloud provider, then a new subclass can be created.
+
+### Persistent Queue for Data Products
+
+_Note: This only applies when a separate process or application is responsible for ingesting data products from the TCS_
+
+
+When a TCS generates data products to be ingested, it's helpful to use a queueing system such as [RabbitMQ](https://www.rabbitmq.com/) to keep track of images to be ingested. By utilizing a simple queuing architecture, if the application responsible for ingesting images ever experiences downtime, the TCS can still queue images to be ingested and the ingestion application will be able to catch up after its outage.
+
+![Ingester queueing scheme](/assets/images/ingester_queueing_scheme.png)
+
+The queue message can be as simple as the location from which to retrieve the file to be ingested, but should contain the minimum amount of information necessary for the ingestion application to successfully ingest the data to the science archive.
+
+```json
+{"filepath": "/path/to/file/on/disk"}
+```
+
+### Retry Scheme for Data Product Ingestion
+
+For added robustness, it is good practice to integrate retry logic into the data product ingestion process. It's for this reason that the OCS Ingester defines a set of Python [exception classes](https://github.com/observatorycontrolsystem/ingester/blob/main/ocs_ingester/exceptions.py) that will be raised in appropriate situations:
+
+* **RetryError** - Exception that is raised when an error happens that can be retried.
+* **BackoffRetryError** - Exception that is raised when an error happens that can be retried with an expontential backoff. For example, networking latency errors that may succeed at a later time.
+* **NonFatalDoNotRetryError** - Exception that is raised when an error happens that should not be retried and is also not a fatal condition.
```

Comment:
maybe give an example of this to see how its different from the DoNotRetryError. I think is if the file already exists in the archive.